lib/internal/modules/esm/module_job.js

```diff
@@ -92,7 +94,18 @@ class ModuleJob {
         this.module.instantiate();
       }
     } catch (e) {
-      decorateErrorStack(e);
+      if (e.message.includes('does not provide an export named')) {
+        const rootURL = this.module.url;
+        const childSpecifier = StringPrototypeMatch(e.message, /'(.*?)'/)[1];
```

Comment:
what happens if this doesn't actually match? couldn't a module have `throw 'does not provide an export named'` in it, and that would break this code? or is this prior to actual user ESM being evaluated?